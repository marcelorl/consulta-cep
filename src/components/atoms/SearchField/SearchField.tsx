import React from 'react'
import styled from 'styled-components'
import InputMask from 'react-input-mask'
import { omit } from 'lodash'

interface ISearchField {
  id: string
  onChange: any
  value: string
}

const SearchFieldWrapper = styled(InputMask)`
  border: 1px solid ${props => props.theme.colors.darkgray};
  border-radius: 5px;
  box-shadow: 0;
  font-size: 1rem;
  padding: 5px 10px;
`

const SearchField = (props: ISearchField) => {
  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    props.onChange(e.target.value)
  }

  return (
    <SearchFieldWrapper
      {...omit(props, 'onChange')}
      mask='99999-999'
      onChange={onChange}
      data-testid='search-field'
      placeholder='01234-567'
      required
    />
  )
}

export default React.memo(SearchField)
