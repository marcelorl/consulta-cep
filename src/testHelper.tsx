import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'

import { Store, useAddress } from './contexts/store'
import { globalTheme } from './globalTheme'
import { IState } from './models'

export const renderWithProviders = (state: IState, ui: any) => {
  const StoreProvider = (props: any) => {
    const [ _, action ] = useAddress()

    return (
      <Store.Provider value={{ state, action }}>
        {props.children}
      </Store.Provider>
    )
  }

  return render(
    <StoreProvider>
      <ThemeProvider theme={globalTheme}>
        {ui}
      </ThemeProvider>
    </StoreProvider>
  )
}

export const renderWithStyledProvider = (ui: any) => {
  return render(
    <ThemeProvider theme={globalTheme}>
      {ui}
    </ThemeProvider>
  )
}
