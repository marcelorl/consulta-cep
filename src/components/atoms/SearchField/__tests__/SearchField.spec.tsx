import React from 'react'
import { cleanup } from 'react-testing-library'

import SearchField from '../SearchField'
import { renderWithStyledProvider } from '../../../../testHelper'

afterEach(cleanup)

describe('#SearchField atom', () => {
  it('renders component', () => {
    const { container } = renderWithStyledProvider(<SearchField id="id" />)

    expect(container).toMatchSnapshot()
  })

  // it('tests onChange is called on typing', () => {
  //   const onChange = jest.fn()
  //   const { container } = render(
  // <ThemeProvider theme={globalTheme}>
  //       <SearchField id="id" onChange={onChange} />
  //     </ThemeProvider>
  //   )
  //
  //   wrapper.simulate('change', { target: { value: '' } })
  //
  //   expect(onChange).toBeCalled()
  // })
})
