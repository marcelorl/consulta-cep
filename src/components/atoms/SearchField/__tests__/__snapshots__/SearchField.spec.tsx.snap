// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SearchField atom renders component 1`] = `
.c0 {
  border: 1px solid #999999;
  border-radius: 5px;
  box-shadow: 0;
  font-size: 1rem;
  padding: 5px 10px;
}

<div>
  <input
    class="c0"
    data-testid="search-field"
    id="id"
    placeholder="01234-567"
    required=""
  />
</div>
`;
